src/pdf_modifier_mcp/web/routes/pdf.py

```
@@ -0,0 +1,200 @@
+        if pages:
+            parts = pages.split("-")
+            if len(parts) == 1:
+                page_range = (int(parts[0]), int(parts[0]))
+            elif len(parts) == 2:
+                page_range = (int(parts[0]), int(parts[1]))
+
+        result = modifier.process(spec, pages=page_range)
+        session_mgr.set_modified_path(session_id, output_path)
+        return result.model_dump()
+    except PDFModifierError as e:
+        raise HTTPException(status_code=400, detail=e.message)
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)